spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpatialKeyIndex.scala

```diff
@@ -69,8 +70,10 @@ class HilbertSpatialKeyIndex(val keyBounds: KeyBounds[SpatialKey], val xResoluti
         BitVectorFactories.OPTIMAL.apply(yResolution)
       )
 
-    bitVectors(0).copyFrom(key.col.toLong)
-    bitVectors(1).copyFrom(key.row.toLong)
+    val col = key.col - minKey.col
```

Comment:
This would break current Hilbert layers! Why is this change in here?

Reply:
yeah, but it's a bug, otherwise this test would fail with a bit overflow error, in a `.copyFrom` function call: 

```
it("Generates a Correct index given a 4x4 square on a 10th zoom level") {
  val keyBounds = KeyBounds(SpatialKey(177, 409), SpatialKey(178, 410))
  val index = HilbertKeyIndexMethod.createIndex(keyBounds)
  index.toIndex(SpatialKey(178, 410)) should be (2)
}
```

Reply:
In addition, it's how a curent temporal `HilbertKeyIndex` works: 

https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala#L88-L89

Reply:
There is also a workaround (not a real fix but at least to make it not failing): 

```scala
     def apply(temporalResolution: Int): KeyIndexMethod[SpaceTimeKey] =
        new KeyIndexMethod[SpaceTimeKey] {
          def createIndex(keyBounds: KeyBounds[SpaceTimeKey]) = {
            val xResolution = math.max(resolution(keyBounds.maxKey.col, keyBounds.minKey.col), 64 - numberOfLeadingZeros(keyBounds.maxKey.col.toLong))
            val yResolution = math.max(resolution(keyBounds.maxKey.row, keyBounds.minKey.row), 64 - numberOfLeadingZeros(keyBounds.maxKey.row.toLong))

             HilbertSpaceTimeKeyIndex(keyBounds, xResolution, yResolution, temporalResolution)
           }
        }
```

Reply:
I remember this, but should this be a separate PR with it's own discussion on fallout on current layers, and how to mitigate any potential damage to current layers?